aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterQueryTest.java

```diff
@@ -225,6 +243,299 @@ public void querySavedDataWithMultipleForeignKeysOfSameType() throws DataStoreEx
+                .build();
+
+        adapter.save(blogOwnerAlpha);
+        adapter.save(blogOwnerBeta);
+        adapter.save(blogOwnerGamma);
+        adapter.save(postByAlpha);
+        adapter.save(postByBeta);
+        adapter.save(postByGamma);
+
+        List<Post2> posts = adapter.query(Post2.class,
+                Where.matches(Post2.BLOG_OWNER.eq(blogOwnerAlphaId)));
+        assertEquals(1, posts.size());
+        assertTrue(posts.contains(postByAlpha));
+
+        posts = adapter.query(Post2.class,
+                Where.matches(Post2.BLOG_OWNER.eq(blogOwnerBetaId)));
+        assertEquals(1, posts.size());
+        assertTrue(posts.contains(postByBeta));
+
+        posts = adapter.query(Post2.class,
+                Where.matches(Post2.BLOG_OWNER.eq(blogOwnerGammaId)));
+        assertEquals(1, posts.size());
+        assertTrue(posts.contains(postByGamma));
+
+    }
+
+    /**
+     * Test that querying the saved item with multiple paths for foreign keys to the same
+     * model along with Multiple conditions correctly forms the where clause and returns expected
+     * data.
+     * @throws DataStoreException On unexpected failure manipulating items in/out of DataStore.
+     */
+    @Test
+    public void querySavedDataWithMultipleJoinPathsWithMultipleConditions() throws DataStoreException {
+
+        final String blogOwnerId = UUID.randomUUID().toString();
+        final String blogId = UUID.randomUUID().toString();
+        final String postId = UUID.randomUUID().toString();
+
+        final BlogOwner3 blogOwner = BlogOwner3.builder()
+                .name("Sample BlogOwner")
+                .id(blogOwnerId)
+                .build();
+
+        final Blog3 blog = Blog3.builder()
+                .name("Sample Blog")
+                .id(blogId)
+                .owner(blogOwner)
+                .build();
+
+        final Post2 post = Post2.builder()
+                .title("Placeholder title")
+                .status(PostStatus.ACTIVE)
+                .rating(5)
+                .id(postId)
+                .blog(blog)
+                .blogOwner(blogOwner)
+                .build();
+
+        adapter.save(blogOwner);
+        adapter.save(blog);
+        adapter.save(post);
+
+        final List<Post2> posts = adapter.query(Post2.class,
+                Where.matches(Post2.BLOG_OWNER.eq(blogOwnerId).and(Post2.RATING.gt(4).
+                        and(Post2.STATUS.eq(PostStatus.ACTIVE)))));
+        assertTrue(posts.contains(post));
+    }
+
+    /**
+     * Test that querying the saved item with multiple paths for foreign keys to the same
+     * model along with Multiple conditions and Custom Primary Key correctly forms the where clause
+     * and returns expected data.
+     * @throws DataStoreException On unexpected failure manipulating items in/out of DataStore.
+     */
+    @Test
+    public void querySavedDataWithMultipleJoinPathsWithMultipleConditionsAndCPK() throws DataStoreException {
```

Comment:
Can you adjust this query to use the composite key: https://docs.amplify.aws/android/build-a-backend/more-features/datastore/customize-primary-keys/#querying-records-with-custom-primary-keys

Reply:
I have added Composite Key for Post in Query.